Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -376,7 +407,7 @@ void UPSWorldSubsystem::ResetSaveGameData()
 // Unlocks all levels of the Progression System
 void UPSWorldSubsystem::UnlockAllLevels()
 {
-	SaveGameInstanceInternal->UnlockAllLevels();
+	SaveGameDataInternal->UnlockAllLevels();
```

Comment:
here and above in `ResetSaveGameData`+`SaveDataAsync`, add `checkf` before pointer usage, if it always has to be valid (so it will crash if not):
```
checkf(SaveGameDataInternal, TEXT("ERROR: [%i] %hs:\n'SaveGameDataInternal' is not created!"), __LINE__, __FUNCTION__);`
SaveGameDataInternal->...
```

Reply:
added ensureMsgf instead of crash: https://github.com/h4rdmol/ProgressionSystem/pull/18/commits/8c35631efb6756e6c34e844e1f08c4a0ffc6cbf1

I also found a bunch of checkf inside the module and not sure if it's okay for a module to perform a crash of a whole game. Shall I replace all checkf with ensureMsgf?